devito/data/data.py

```diff
@@ -132,16 +142,82 @@ def _global(self, glb_idx, decomposition):
         ret._is_distributed = any(i is not None for i in decomposition)
         return ret
 
+    def _check_idx(func):
+        """Check if __getitem__/__setitem__ may require communication across MPI ranks."""
+        @wraps(func)
+        def wrapper(data, *args, **kwargs):
+            glb_idx = args[0]
+            if len(args) > 1 and isinstance(args[1], Data) \
+                    and args[1]._is_mpi_distributed:
+                comm_type = PARALLEL
+            elif data._is_mpi_distributed:
+                for i in as_tuple(glb_idx):
+                    if isinstance(i, slice) and i.step is not None and i.step < 0:
+                        comm_type = PARALLEL
```

Comment:
I think I still prefer a boolean "needs_comm" over PARALLEL vs SERIAL, which is not btw what the docstring claims to check?

Reply:
Docstring says:
```
"""Check if __getitem__/__setitem__ may require communication across MPI ranks."""
```
?

So you want a boolean `needs_comm` and remove the `class COMM_TYPE(Tag)`? If so, this is exactly how the code was previously, albeit with different names.